### PR TITLE
Create server discovery endpoints

### PR DESCRIPTION
**Pull request by EduPonz**
This PR adds the creation of PDP and EDP discovery endpoints. Merge after #1391

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2629/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/45/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2446/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2639/)

File: src/cpp/rtps/builtin/discovery/endpoint/EDPServer2.hpp [hunk 1]
```diff
@@ -104,6 +111,12 @@ class EDPServer2 : public aux::EDPSimple
      */
     virtual bool createSEDPEndpoints() override;
 
+    //! EDP publications writer filter
+    IReaderDataFilter* edp_publications_filter_;
+
+    //! EDP subscriptions writer filter
+    IReaderDataFilter* edp_subscriptions_filter_;
+
```
Comment (MiguelBarro) [inline src/cpp/rtps/builtin/discovery/endpoint/EDPServer2.hpp:119]: Why should we keep the interfaces references? Once passed to the Writers they are no longer useful.

File: src/cpp/rtps/builtin/discovery/participant/PDPServer2.hpp [hunk 2]
```diff
@@ -124,12 +126,18 @@ class PDPServer2 : public aux::PDP
...
+
 private:
 
     /**
      * TimedEvent for server synchronization
      */
     DServerEvent2* mp_sync;
+
+    //! PDP writer filter
+    IReaderDataFilter* pdp_filter_;
```
Comment (MiguelBarro) [inline src/cpp/rtps/builtin/discovery/participant/PDPServer2.hpp:140]: again, Why keep the interface reference?

**Review by MiguelBarro — CHANGES_REQUESTED** [review]
Open to discussion.

File: include/fastdds/rtps/builtin/discovery/participant/PDPServer.h [hunk 3]
```diff
@@ -61,6 +62,8 @@ class PDPServer : public PDP
     // TODO: remove when the Writer API issue is resolved
     std::map<GUID_t, ReaderProxyData> clients_;
 
+    fastdds::rtps::DiscoveryDataBase discovery_database_;
```
[the remark this answers is not in the document]
Comment (EduPonz) [reply include/fastdds/rtps/builtin/discovery/participant/PDPServer.h]: done

File: src/cpp/rtps/builtin/discovery/endpoint/EDPServer2.hpp (hunk 1 above)
Comment (EduPonz) [reply src/cpp/rtps/builtin/discovery/endpoint/EDPServer2.hpp]: You're right!

File: src/cpp/rtps/builtin/discovery/participant/PDPServer2.hpp (hunk 2 above)
Comment (EduPonz) [reply src/cpp/rtps/builtin/discovery/participant/PDPServer2.hpp]: done

**Review by MiguelBarro — APPROVED** [review]
LGTM

**Pull request merged**